### PR TITLE
CLN: remove _combine_series_frame

### PR DESCRIPTION
**Pull request by jbrockmendel**
Following this we can consolidate some dispatch_to_series calls, saving that for a separate step for clarity of exposition.

### CODE REVIEW

**Comment by jreback** [comment]
lgtm ping on green.

File: pandas/core/computation/expressions.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def evaluate(op, a, b, use_numexpr: bool = True):
     use_numexpr : bool, default True
         Whether to try to use numexpr.
     """
-    op_str = _op_str_mapping.get(op, None)
+    op_str = _op_str_mapping[op]
     if op_str is not None:
```
Comment (topper-123) [inline pandas/core/computation/expressions.py:227]: Isnt this always true, so you can drop the if statement now?

Comment (jbrockmendel) [reply pandas/core/computation/expressions.py:227]: it is not always true, some entries in _op_str_mapping give None

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**